src/Compilers/CSharp/Test/Emit3/Semantics/ExtensionOperatorsTests.cs

```diff
@@ -572,6 +574,1194 @@ struct S1
+
+        [Fact]
+        public void Unary_030_Consumption_BadOperand()
+        {
+            var src = $$$"""
+public static class Extensions1
+{
+    extension(object)
+    {
+        public static object operator +(object x)
+        {
+            System.Console.Write("operator1");
+            return x;
+        }
+    }
+}
+
+public struct S1
+{}
+class Program
+{
+    static void Main()
+    {
+        _ = +null;
+        _ = +default;
+        _ = +new();
+        _ = +(() => 1);
+    }
+}
+""";
+
+            var comp = CreateCompilation(src, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics(
+                // (19,13): error CS8310: Operator '+' cannot be applied to operand '<null>'
+                //         _ = +null;
+                Diagnostic(ErrorCode.ERR_BadOpOnNullOrDefaultOrNew, "+null").WithArguments("+", "<null>").WithLocation(19, 13),
+                // (20,14): error CS8716: There is no target type for the default literal.
+                //         _ = +default;
+                Diagnostic(ErrorCode.ERR_DefaultLiteralNoTargetType, "default").WithLocation(20, 14),
+                // (21,14): error CS8754: There is no target type for 'new()'
+                //         _ = +new();
+                Diagnostic(ErrorCode.ERR_ImplicitObjectCreationNoTargetType, "new()").WithArguments("new()").WithLocation(21, 14),
+                // (22,13): error CS0023: Operator '+' cannot be applied to operand of type 'lambda expression'
+                //         _ = +(() => 1);
+                Diagnostic(ErrorCode.ERR_BadUnaryOp, "+(() => 1)").WithArguments("+", "lambda expression").WithLocation(22, 13)
+                );
+        }
+
+        [Fact]
+        public void Unary_031_Consumption_BadReceiver()
+        {
+            var src = $$$"""
+public static class Extensions1
+{
+    extension(__arglist)
+    {
+        public static object operator +(object x)
+        {
+            return x;
+        }
+    }
+}
+
+class Program
+{
+    static void Main()
+    {
+        var s1 = new object();
+        _ = +s1;
+    }
+}
+""";
+
+            var comp = CreateCompilation(src, options: TestOptions.DebugExe);
+            comp.VerifyDiagnostics(
+                // (3,15): error CS1669: __arglist is not valid in this context
+                //     extension(__arglist)
+                Diagnostic(ErrorCode.ERR_IllegalVarArgs, "__arglist").WithLocation(3, 15),
+                // (5,39): error CS9551: The parameter of a unary operator must be the extended type.
+                //         public static object operator +(object x)
+                Diagnostic(ErrorCode.ERR_BadExtensionUnaryOperatorSignature, "+").WithLocation(5, 39),
+                // (17,13): error CS0023: Operator '+' cannot be applied to operand of type 'object'
+                //         _ = +s1;
+                Diagnostic(ErrorCode.ERR_BadUnaryOp, "+s1").WithArguments("+", "object").WithLocation(17, 13)
+                );
+        }
```

Comment:
Consider testing consumption of operator defined in metadata image (I think I only saw compilation references)